### PR TITLE
build: add alpine support to matrix builds

### PR DESCRIPTION
**Pull request by itzg**

### CODE REVIEW

File: build/run.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+set -e
+
+distro=$(cat /etc/os-release | grep -E "^ID=" | cut -d= -f2 | sed -e 's/"//g')
+
+"$(dirname "$0")/${distro}/$1".sh
```
Comment (shotah) [inline build/run.sh:7]: Nice folder shim!

File: tests/setuponlytests/test.sh [hunk 2]
```diff
@@ -40,7 +40,7 @@ $logs
     echo "${folder} PASS"
   fi
 
-  docker-compose down -v --remove-orphans
+  docker-compose down -v --remove-orphans > /dev/null
```
Comment (shotah) [inline tests/setuponlytests/test.sh:43]: Do we need the -v if we are just outputting to Null anyways? Notting seeing -v referenced, but pretty sure it's short for --verbose. https://docs.docker.com/compose/reference/

Comment (itzg) [reply tests/setuponlytests/test.sh:43]: It's for removing any names volumes

https://docs.docker.com/compose/reference/down/

File: scripts/start-utils [hunk 3]
```diff
@@ -237,11 +237,15 @@ function extract() {
   esac
 }
 
+function getDistro() {
+  cat /etc/os-release | grep -E "^ID=" | cut -d= -f2 | sed -e 's/"//g'
```
[the remark this answers is not in the document]
Comment (itzg) [reply scripts/start-utils:241]: That particular one apparently came from Github Copilot via a recent PR

**Comment by itzg** [comment]
@shotah Thanks for taking a look at the PR BTW

File: tests/setuponlytests/test.sh (hunk 2 above)
Comment (shotah) [reply tests/setuponlytests/test.sh:43]: Doh, I think I remember that now. :) Glad you caught it!

**Comment by shotah** [comment]
> @shotah Thanks for taking a look at the PR BTW

Hey I'm excited for alpine to be reintroduced. This does updated it for Java17 and Latest correct?

**Comment by itzg** [comment]
Yep, it became super easy to add back an alpine variant for Java 17. It's also going to make it much easier to add GraalVM, which has a base image on CentOS.

**Pull request merged**